### PR TITLE
Update doc replace to vite

### PR DESCRIPTION
**Pull request by gento-ogane**
#11305 

In [Vite](https://vitejs.dev/), default file name is `main.jsx` instead of `index.js`, so I replaced `index.js` to `main.jsx`.

### Checklist:
- [ ] If this PR contains changes to the library itself (not necessary for e.g. docs updates), please include a changeset (see [CONTRIBUTING.md](https://github.com/apollographql/apollo-client/blob/main/CONTRIBUTING.md#changesets))
- [ ] If this PR is a new feature, please reference an issue where a consensus about the design was reached (not necessary for small changes)
- [ ] Make sure all of the significant new logic is covered by tests

### CODE REVIEW

**Comment by jerelmiller** [comment]
Thanks for the contribution @gento-ogane! We'll do our best to get to this PR as soon as we can. A couple of our team is at conferences this week so we may be a bit slow to get to this. Apologies in advance!

File: docs/shared/link-chain.mdx [hunk 1]
```diff
@@ -1,4 +1,4 @@
-```js title="index.js"
+```jsx title="main.jsx""
```
Comment (illyakurochkin) [inline docs/shared/link-chain.mdx:1]: Why did you change it to `jsx`?

Comment (jerelmiller) [inline docs/shared/link-chain.mdx:1]: I'd prefer to revert the changes to all the non-introductory docs back to `index.js`. I think it makes sense in the context of the vite-specific documentation, but for the rest of it, `index.js` is a fairly common, tool agnostic name.

On a separate note, it looks like you have an extra `"` at the end here 🙂

File: docs/source/get-started.mdx [hunk 2]
```diff
@@ -34,15 +34,15 @@ Our example application will use the [FlyBy GraphQL API](https://flyby-router-de
 
 With our dependencies set up, we can now initialize an `ApolloClient` instance.
 
-In `index.js`, let's first import the symbols we need from `@apollo/client`:
+In `main.jsx`, let's first import the symbols we need from `@apollo/client`:
 
-```js title="index.js"
+```jsx title="main.jsx"
```
Comment (jerelmiller) [inline docs/source/get-started.mdx:39]: I think these changes in the context of this doc are totally fine! Your justification that its a Vite default makes sense 🙂

**Review by jerelmiller — CHANGES_REQUESTED** [review]
Hey @gento-ogane 👋 

I am so sorry it took me so long to get back to you on this PR! I had some time to sit down and run through the getting started to make sure everything still worked/made sense and all looks good!

The only change I'd request you make is reverting the changes in type/filenmae to all docs except the "Getting started" page. Once you're done there, I'd be happy to approve this PR and get this merged. Thanks so much for your patience 🙏

**Review by jerelmiller — APPROVED** [review]
Thanks a bunch for your contribution @gento-ogane! I'm sure new newcomers will really appreciate this change 🙂

**Comment by gento-ogane** [comment]
@jerelmiller 
Sorry I did not notice your reply...
Thanks for your review and commit  and merge!!

**Comment by jerelmiller** [comment]
You're very welcome!

**Pull request merged**